channel.go

```diff
@@ -402,7 +415,9 @@ func (ch *Channel) ListenAndServe(hostPort string) error {
 		return errAlreadyListening
 	}
 
-	l, err := net.Listen("tcp", hostPort)
+	lc := net.ListenConfig{}
```

Comment:
Note: Yarpc does not hit this method for tchannel servers, it creates a listener of its own and passes into `Serve`. You add this under `EnableMPTCP` flag

Reply:
yea I'm aware of that. But for the sake of completeness of this feature, I think we need to do this change.

Reply:
On my understanding, if a `Channel` is passed in as options to create a `ChannelTransport`, yarpc will use ListenAndServe: https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/channel_transport.go#L168

Reply:
we actually use `Transport` in yarpc: https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/transport.go#L285. It uses `Serve()`